spec/template.dd

```diff
@@ -818,60 +818,124 @@ $(GNAME TemplateSequenceParameter):
+        called a type sequence or $(I TypeSeq).)
+    $(LI An $(I AliasSeq) whose elements consist entirely of values is
+        called a value sequence or $(I ValueSeq).)
+    $(LI `typeof` can be used on a *ValueSeq* to obtain a *TypeSeq*.)
     )
 
-    $(P An $(I AliasSeq) can be used as an argument list to instantiate
-        another template, or as the list of parameters for a function.)
+    $(P The elements of an $(I AliasSeq) are automatically expanded
+        when it is referenced in a declaration or expression.
+        An $(I AliasSeq) can be used as arguments to instantiate a
+        template. A *ValueSeq* can be used as arguments to call a
+        function:)
 
         $(SPEC_RUNNABLE_EXAMPLE_RUN
         ---
-        import std.stdio;
+        import std.stdio : writeln;
 
-        template print(args...)
+        template print(args...) // args must be a ValueSeq
         {
             void f()
             {
-                writeln("args are ", args); // args is a ValueSeq
+                writeln("args are ", args);
             }
         }
 
-        template write(Args...) // Args is a TypeSeq
+        void main()
+        {
+            print!(1,'a',6.8).f(); // prints: args are 1a6.8
+        }
+        ---
+        )
+
+        $(P A *TypeSeq* can be used to declare parameters for a function:)
+
+        $(SPEC_RUNNABLE_EXAMPLE_RUN
+        ---
+        import std.stdio : writeln;
+
+        template print(Types...) // Types must be a TypeSeq
         {
-            void f(Args args) // args is a ValueSeq
+            void f(Types args) // args is a ValueSeq
             {
                 writeln("args are ", args);
             }
         }
 
         void main()
         {
-            print!(1,'a',6.8).f();                    // prints: args are 1a6.8
-            write!(int, char, double).f(1, 'a', 6.8); // prints: args are 1a6.8
+            print!(int, char, double).f(1, 'a', 6.8); // prints: args are 1a6.8
         }
         ---
         )
 
-    $(P The number of elements in an $(I AliasSeq) can be retrieved with
-        the $(D .length) property. The $(I n)th element can be retrieved
-        by indexing the $(I AliasSeq) with [$(I n)],
-        and sub-sequences are denoted by the slicing syntax.
+    $(P A *TypeSeq* can similarly be used to
+        $(DDSUBLINK articles/ctarguments, type-seq-instantiation, declare variables).
+        Parameters or variables declared with a *TypeSeq* are called an
+        *lvalue sequence*.)
+
+$(H4 $(LNAME2 seq-ops, Sequence Operations))
+
+    $(UL
+    $(LI The number of elements in an $(I AliasSeq) can be retrieved with
+        the $(D .length) property.)
+    $(LI The $(I n)th element can be retrieved by
+        $(DDSUBLINK spec/expression, index_expressions, indexing) an
+        $(I AliasSeq) with `Seq[n]`. Indexes must be known at compile-time.
+        The result is an lvalue when the element is a variable.)
+    $(LI Sub-sequences can be obtained using
```

Comment:
"Sub-sequence" is not defined as far as I can see, I would not give it a name and say that slicing a sequence produces a new sequence with a subset of the elements of the original sequence.

Reply:
OK, I've now used that wording.